RTech.CoffeeMachine.Api/BrewCoffee/Filters/BrewStatusUnavailableFilter.cs

```diff
@@ -0,0 +1,40 @@
+﻿using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.Filters;
+using Microsoft.Extensions.Caching.Memory;
+using RTech.CoffeeMachine.Api.BrewCoffee.Controllers;
+
+namespace RTech.CoffeeMachine.Api.BrewCoffee.Filters;
+
+public class BrewStatusUnavailableFilter : ActionFilterAttribute
+{
+    // Interviewer note: MemoryCache from Microsoft.Extensions.Caching.Abstractions is thread safe
+    private readonly IMemoryCache _cache;
+    public static readonly string CACHE_KEY_NUM_REQUESTS = $"{nameof(CoffeeMachineController.GetBrewStatus)}:numRequests";
```

Comment:
- Depending on the company/team, they may already have a naming convention for cache keys.
- For instance, Redis (which is distributed across multiple app services) might use `SERVICE.ENTITY.KEY` (e.g.: `CoffeeMachine.BrewStatus.NumRequests`).
- Here i've just used a simple key that ensures uniqueness going forward, but if a pre-existing convention exists, i'll adopt the established convention instead